### ISSUE TITLE
Adding batch operation APIs for PinotFS

### ISSUE THREAD

**Issue by walterddr** [apache/pinot#8064]
Currently PinotFS doesn't support batch operation API extensions easily. 

For some usage such as SegmentDeletionManager. once has to iterate over all files checking for deletion and then do the actual delete. but for many of the cloud FS, there's more efficient batch APIs to reduce significantly on the remote request overheads. 

Propose to
1. add `delete` / `copy` / `move` `(List<URI> segmentUris, ...)` API to its single URI variant.
2. make default implementation for them to fall back to looping each individually.

**Comment by Jackie-Jiang** [comment]
Discussed offline, and another issue for remote FS is that `move` might not be an cheap operation (e.g. S3 move is actually copy + delete). To solve this, we should add a config to allow directly deleting the segment without moving it to a different directory.

**Comment by mcvsubbu** [comment]
> 

Do you mean not saving the deleted segments? I believe we already have a config there.

**Comment by Jackie-Jiang** [comment]
@mcvsubbu Which config are you referring to?
I find a config for the retention days before deleting the segment from the deleted segment dir, but didn't find one to skip moving the segment to the deleted segment dir. In `SegmentDeletionManager.removeSegmentFromStore()`, segment is always moved.

**Comment by walterddr** [comment]
for this issue it is more the multiple-round-trip per segment when we operate on the entire table. 

Even if the deleteSegment does the deletion only without the move. on a large enough table it will still take 10s of minutes to complete the deletion since it requires pinot to issue `delete(URI segmentUri)` sequentially segment after segment instead of leveraging the underlying PinotFS impl, which might have a much more efficient way to batch operator on a list of segment URIs

**Comment by richardstartin** [comment]
> for this issue it is more the multiple-round-trip per segment when we operate on the entire table.
> 
> Even if the deleteSegment does the deletion only without the move. on a large enough table it will still take 10s of minutes to complete the deletion since it requires pinot to issue `delete(URI segmentUri)` sequentially segment after segment instead of leveraging the underlying PinotFS impl, which might have a much more efficient way to batch operator on a list of segment URIs

This raises the question of how the state of the deletion operation is maintained, currently it is managed by the client. Some databases do the wrong thing and let you delete as many rows as possible before a timeout is reached because they treat this as a stateless operation.

**Comment by walterddr** [comment]
> This raises the question of how the state of the deletion operation is maintained, currently it is managed by the client. Some databases do the wrong thing and let you delete as many rows as possible before a timeout is reached because they treat this as a stateless operation.

I dont think it is currently "managed" by the client. if some segments were not deleted (moved), it simply just logs a warning and move on to the next segment, it doesn't even return it back to the client.
